lib/cli/src/commands/init.rs

```diff
@@ -0,0 +1,503 @@
+    /// Initialize an empty wapm.toml
+    #[clap(long, group = "crate-type")]
+    pub empty: bool,
+    /// Force overwriting the wapm.toml, even if it already exists
+    #[clap(long)]
+    pub overwrite: bool,
+    /// Don't display debug output
+    #[clap(long)]
+    pub quiet: bool,
+    /// Namespace to init with, default = current logged in user or _
+    #[clap(long)]
+    pub namespace: Option<String>,
+    /// Package name to init with, default = Cargo.toml name or current directory name
+    #[clap(long)]
+    pub package_name: Option<String>,
+    /// Version of the initialized package
+    #[clap(long)]
+    pub version: Option<semver::Version>,
+    /// If the `manifest-path` is a Cargo.toml, use that file to initialize the wapm.toml
+    #[clap(long)]
+    pub manifest_path: Option<PathBuf>,
+    /// Add default dependencies for common packages
+    #[clap(long, value_enum)]
+    pub template: Option<Template>,
+    /// Include file paths into the target container filesystem
+    #[clap(long)]
+    pub include: Vec<String>,
+    /// Directory of the output file name. wasmer init will error if the target dir
+    /// already contains a wasmer.toml. Also sets the package name.
+    #[clap(name = "PACKAGE_PATH")]
+    pub out: Option<PathBuf>,
+}
+
+/// What template to use for the initialized wasmer.toml
+#[derive(Debug, PartialEq, Eq, Copy, Clone, clap::ValueEnum)]
+pub enum Template {
+    /// Add dependency on Python
+    Python,
+    /// Add dependency on JS
+    Js,
+}
+
+#[derive(Debug, PartialEq, Copy, Clone)]
+enum BinOrLib {
+    Bin,
+    Lib,
+    Empty,
+}
+
+// minimal version of the Cargo.toml [package] section
+#[derive(Debug, Clone)]
+struct MiniCargoTomlPackage {
+    name: String,
+    version: semver::Version,
+    description: Option<String>,
+    homepage: Option<String>,
+    repository: Option<String>,
+    license: Option<String>,
+    readme: Option<PathBuf>,
+    license_file: Option<PathBuf>,
+    #[allow(dead_code)]
+    workspace_root: PathBuf,
+    #[allow(dead_code)]
+    build_dir: PathBuf,
+}
+
+static WASMER_TOML_NAME: &str = "wasmer.toml";
+
+impl Init {
+    /// `wasmer init` execution
+    pub fn execute(&self) -> Result<(), anyhow::Error> {
+        let bin_or_lib = self.get_bin_or_lib()?;
+
+        // See if the directory has a Cargo.toml file, if yes, copy the license / readme, etc.
+        let manifest_path = match self.manifest_path.as_ref() {
+            Some(s) => s.clone(),
+            None => {
+                let cargo_toml_path = self
+                    .out
+                    .clone()
+                    .unwrap_or_else(|| std::env::current_dir().unwrap())
+                    .join("Cargo.toml");
+                cargo_toml_path
+                    .canonicalize()
+                    .unwrap_or_else(|_| cargo_toml_path.clone())
+            }
+        };
+
+        let cargo_toml = if manifest_path.exists() {
+            Some(parse_cargo_toml(&manifest_path)?)
```

Comment:
We should notify the user (via print) that the cli is generating the wasmer.toml using the settings from cargo

Reply:
Done in https://github.com/wasmerio/wasmer/pull/3432/commits/dd470adeffc4fd80cd1b29f99746d276c1088c42